File: src/main/java/com/preeti/campushub/service/CourseService.java
package com.preeti.campushub.service;

import java.util.List;

import org.springframework.data.domain.Page;

import com.preeti.campushub.dto.course.CourseRequest;
import com.preeti.campushub.dto.course.CourseResponse;
import com.preeti.campushub.dto.student.StudentResponse;

public interface CourseService {

    CourseResponse createCourse(CourseRequest request);

    List<CourseResponse> getAllCourses();

    Page<CourseResponse> getCourses(
        int page,
        int size,
        String sortBy,
        String direction);

    List<CourseResponse> searchCourses(String keyword);

    List<CourseResponse> filterCourses(
        Long departmentId,
        Long facultyId);

    CourseResponse getCourseById(Long id);

    CourseResponse updateCourse(Long id, CourseRequest request);

    void deleteCourse(Long id);

    List<CourseResponse> getMyCourses();

    List<StudentResponse> getStudentsByCourse(Long courseId);

}

File: campushub-frontend/src/services/courseService.js
import api from "../api/axiosConfig";

export const getAllCourses = async () => {
    const response = await api.get("/courses");
    return response.data.data;
};

export const createCourse = async (course) => {
    const response = await api.post("/courses", course);
    return response.data.data;
};

export const updateCourse = async (id, course) => {
    const response = await api.put(`/courses/${id}`, course);
    return response.data.data;
};

export const deleteCourse = async (id) => {
    const response = await api.delete(`/courses/${id}`);
    return response.data.data;
};

export const getMyCourses = async () => {
    const response = await api.get("/courses/my-courses");
    return response.data.data;
};

export const getStudentsByCourse = async (courseId) => {
    const response = await api.get(`/courses/${courseId}/students`);
    return response.data.data;
};